lib/src/algorithms.dart

```diff
@@ -32,10 +32,10 @@ int _comparableBinarySearch/*<T extends Comparable<T>>*/(
 /// the objects.
 ///
 /// Returns -1 if [value] is not in the list by default.
-int binarySearch/*<T extends Comparable<T>>*/(
+int binarySearch/*<T>*/(
     List/*<T>*/ sortedList, /*=T*/ value, { int compare(/*=T*/ a, /*=T*/ b) }) {
   if (compare == null) {
-    return _comparableBinarySearch(sortedList, value);
+    return _comparableBinarySearch(sortedList as List<Comparable>, value as Comparable);
```

Comment:
At the moment, yes.  The comment-based generic method syntax is only turned on by default in the analyzer when in strong mode (it may be possible to turn it on in spec-mode, not sure).  The analyzer also supports the actual proposed syntax (basically remove the /**/ from the comment based syntax) behind a flag.  Support for generic methods in other tools (dart2js, VM) is further behind, hence the comment based syntax.